source/material_model/steinberger.cc

```diff
@@ -578,71 +454,86 @@ namespace aspect
-              out.thermal_expansion_coefficients[i] = thermal_expansion_coefficient (in.temperature[i], in.pressure[i], in.composition[i], in.position[i]);
-              out.specific_heat[i]                  = specific_heat                 (in.temperature[i], in.pressure[i], in.composition[i], in.position[i]);
-            }
-          out.thermal_conductivities[i]         = thermal_conductivity          (in.temperature[i], in.pressure[i], in.composition[i], in.position[i]);
-          out.compressibilities[i]              = compressibility               (in.temperature[i], in.pressure[i], in.composition[i], in.position[i]);
+          out.thermal_conductivities[i] = thermal_conductivity_value;
           out.entropy_derivative_pressure[i]    = 0;
           out.entropy_derivative_temperature[i] = 0;
           for (unsigned int c=0; c<in.composition[i].size(); ++c)
             out.reaction_terms[i][c]            = 0;
 
-          // fill seismic velocity outputs if they exist
-          if (SeismicAdditionalOutputs<dim> *seismic_out = out.template get_additional_output<SeismicAdditionalOutputs<dim> >())
+          // The following lines take the appropriate averages of the
+          // thermodynamic moaterial properties
+          std::vector<double> specific_heats(material_lookup.size());
+          std::vector<double> densities(material_lookup.size());
+          std::vector<double> compressibilities(material_lookup.size());
+          std::vector<double> thermal_expansivities(material_lookup.size());
+
+          for (unsigned int j=0; j<material_lookup.size(); ++j)
             {
-              seismic_out->vp[i] = seismic_Vp(in.temperature[i], in.pressure[i], in.composition[i], in.position[i]);
-              seismic_out->vs[i] = seismic_Vs(in.temperature[i], in.pressure[i], in.composition[i], in.position[i]);
+              densities[j] = material_lookup[j]->density(in.temperature[i],pressure);
+              compressibilities[j] = material_lookup[j]->dRhodp(in.temperature[i],pressure)/densities[j];
+
+              if (!latent_heat)
+                {
+                  thermal_expansivities[j] = material_lookup[j]->thermal_expansivity(in.temperature[i],pressure);
+                  specific_heats[j] = material_lookup[j]->specific_heat(in.temperature[i],pressure);
+                }
             }
-        }
 
-      // fill phase volume outputs if they exist
-      if (NamedAdditionalMaterialOutputs<dim> *phase_volume_fractions_out = out.template get_additional_output<NamedAdditionalMaterialOutputs<dim> >())
-        fill_phase_volume_fractions(in, phase_volume_fractions_out);
+          out.densities[i] = MaterialUtilities::average_value(volume_fractions[i], densities, MaterialUtilities::arithmetic); // volume fraction averaging of densities
+          out.compressibilities[i] = MaterialUtilities::average_value(volume_fractions[i], compressibilities, MaterialUtilities::arithmetic); // volume averaging of isothermal compressibility
```

Comment:
please put the comments on an extra line, or remove them, I think the lines are reasonably self explanatory (or the comments do at least not add much information)

Reply:
I've now put modified comments on extra lines. They are there as reinforcement (in case someone thinks the choice is arbitrary).